source/common/version/api_version.cc

```diff
@@ -0,0 +1,32 @@
+#include "common/version/api_version.h"
+
+#include <string>
+
+#include "common/common/fmt.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+
+const ApiVersion& ApiVersionInfo::apiVersion() { return api_version; }
+
+const ApiVersion& ApiVersionInfo::oldestApiVersion() {
+  static const auto* result = new ApiVersion(computeOldestApiVersion(apiVersion()));
+  return *result;
+}
+
+ApiVersion ApiVersionInfo::computeOldestApiVersion(const ApiVersion& latest_version) {
+  // Envoy supports up to 2 most recent minor versions. Therefore if the latest
+  // API version "X.Y.Z", Envoy's oldest API version is "X.Y-1.0".
```

Comment:
This is assuming we aren't going over major version, but there are no plans for that any time soon anyway.

Reply:
The current implementation supports only a range of API versions in the same major version (this is also true for the range that will be sent to the management server).
This can be clarified by adding a comment to `computeOldestApiVersion` (and to `oldestApiVersion`) `get and/or to change the methods name to reflect that this is oldest API version with the same major version.
Do you think anything else should be done here?